src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -170,13 +170,20 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
 
         if (isThreeDSecureRequiredError) {
             const clientSecret = error.body.three_ds_result.token;
-            const result = await this._getStripeJs().handleCardAction(clientSecret);
+            let result;
+            let needsConfirm = false;
+            try {
+                result = await this._getStripeJs().handleCardAction(clientSecret);
+                needsConfirm = true;
+            } catch (errorCardAction) {
```

Comment:
What scenario causes `isThreeDSecureRequiredError` to return `true`

and

`handleCardAction` to throw (indicating that 3DS is _not_ needed)?

Reply:
That first scenario is when try to complete the payment  and stripe response back that the payment needs a 3ds auth, bigpay send  back to the front a error with the code  three_d_secure_required and setup isThreeDSecureRequiredErrror as true.

We use the handleCardAction to show the modal for 3ds auth to the shopper, but that will Throw a error is the payment was previously updated on the back end, in that case we use confirmCardPayment to show the the 3ds auth modal.

The only way that 3ds in not needed is when the card do not suppport 3ds.

Reply:
Sorry, I'm not sure that I understand.

> but that will Throw a error is the payment was previously updated on the back end

What update is our back end making which will cause `handleCardAction` to throw?

Is it something we can avoid doing on the back end, rather than trying to handle here?

Reply:
There are two flows, the first one (this is the actual):
In the initialization we create a payment intent to get the client token, when place a order in the back a new payment intent is created with the payment data and stripe tell us that need a 3ds auth and send the three_d_secure_required error to the front to display the 3ds auth modal via handleCardAction to the shopper the shopper authorize and complete the payment.

In the second flow (added in this PR):
We get the client token from the payment intent on the initializtion step, when place the order we send back the client token to the back to get the previous payment intent instead to create a new one and update it with the payment data, in the response for that update stripe tell us that the payment intent needs no be confirmed and we send the confirm to stripe and after stripe response that the payment intent needs a 3ds auth and send three_d_secure_required to the front, in the front we use handleCardAction to show the 3ds modal, but in this case is when handleCardAction throws a exception that indacates whe need to use confirCardPayment to show the 3ds modal.

I support the 2 flows because I need this changes to be merged and hit production before the bigpay one

Reply:
Am I right in understanding that we're wanting to deploy a breaking change to Bigpay, to change how it handles Stripe V3 requests?

And to cover that change, you're altering checkout-sdk to work with both the old _and_ new flows?
(I assume with the idea of removing the old flow after the Bigpay change has been deployed)

Reply:
That's correct

Reply:
The problem with that rollout is that consumers of existing older `checkout-sdk` versions will suddenly suffer from a broken Stripe V3 experience after the changes to Bigpay are deployed.

Typically we do not want to make breaking API changes for that reason. We want to maintain backwards compatibility as much as possible.

As such, the preferred rollout would be:

1. Keep the current `checkout-sdk` Stripe strategy and current Bigpay API functionality
2. Create a new and seperate Stripe strategy which consumes new and and seperate Bigpay API functionality
3. Work to migrate existing Stripe users from the old to the new version
4. Remove the old version functionality from both `checkout-sdk` and Bigpay when everyone is safely migrated

(Sound right to @bigcommerce/checkout ?)

Reply:
@leeBigCommerce  this changes support the current and new functionality of bigpay, but I need this changes be on production before the bigpay ones

Reply:
Maybe I'm missing something 😄 

1. If Bigpay already works the new and old way
2. Then we update Checkout-SDK (with this PR) to work the new _and_ old way
3. We release this updated Checkout-SDK version
4. Then we remove the old way from Bigpay 
5. All merchants who are _not_ on the latest Checkout-SDK version (#3 on this list) will no longer be able to use Stripe

Are the Bigpay changes backwards compatible in a way that I'm not understanding?
Or do we have a different solution for merchants using Stripe on pre-existing Checkout-SDK versions?

Reply:
let me try to explain better

The new way is trigger when the front send back to  bigpay the stripe client token ( this functionality is already in checkout SDK), but actually is ignored by bigpay the code that process the new way is in this PR https://github.com/bigcommerce/bigpay/pull/4135 but It can't be merged because it will fail with 3ds card, that problem is resolved by this PR for that reason this changes support the old and new.

Basically the step are
 1. Merge check sdk with old and new ways
 2. Wait th sdk chagnges reach production
 3. Merge the bigpay changes
 4. After that we can remove the old ways in sdk and bigpay.
 
all of this was caused because previously  merged a two PRs one in sdk that send back the client token and one in bigpay that update the payment intent related to tht client token, but we got some error  with 3ds cards and decided to revert the bigpay one and preserve the sdk one,this because if the bigpay changes are not ther the client token send it by the front is ignored and the old way is executed.

Reply:
After your 4 steps, will _past_ versions of Checkout-SDK still have working Stripe V3 flows?

Reply:
If we want to support it, we need to omit the step  4 and preserve the 2 flows